From dc8f929a2c08f1c72b0a82f8addf88bdf0c354a5 Mon Sep 17 00:00:00 2001
From: Richard Hulme <peper03@mythtv.org>
Date: Sun, 26 Oct 2014 19:12:38 +0100
Subject: [PATCH] Make logic in RingBuffer::WaitForReadsAllowed check whether
 the timeout actually set in timeout_ms has been reached
 rather than assuming timeout_ms is always going to be 10
 seconds.

---
 mythtv/libs/libmythtv/ringbuffer.cpp |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index 35044ab..98682a5 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -1280,10 +1280,11 @@ bool RingBuffer::WaitForReadsAllowed(void)
         }
         count++;
     }
-    if (t.elapsed() > 10000)
+    if (t.elapsed() >= timeout_ms)
     {
         LOG(VB_GENERAL, LOG_ERR, LOC +
-            "Took more than 10 seconds to be allowed to read, aborting.");
+            QString("Took more than %1 seconds to be allowed to read, aborting.")
+            .arg(timeout_ms / 1000));
         return false;
     }
     return check;
-- 
1.7.10.2

